charts/argo-cd/values.yaml

```diff
@@ -313,6 +313,61 @@ configs:
       #   ...
       #   -----END CERTIFICATE-----
 
+  # ConfigMap for Config Management Plugins
+  # Ref: https://argo-cd.readthedocs.io/en/stable/operator-manual/config-management-plugins/
+  cmp:
+    # -- Create the argocd-cmp-cm configmap
+    create: true
+
+    # -- Annotations to be added to argocd-cmp-cm configmap
+    annotations: {}
+
+    # -- Plugin yaml files to be added to argocd-cmp-cm
+    plugins: {}
+      # --- First plugin
+      # my-plugin: |
+      #   ---
+      #   apiVersion: argoproj.io/v1alpha1
```

Comment:
This is just an idea but plugin needs to be unique in Argo CD instance. Do we need to template this as full spec? I would consider putting this into CM template itself and just keep it simple:

```yaml
my-plugin:
  init:
    command: [sh]
    args: [-c, 'echo "Initializing..."' ]
<etc.>
``` 

and inside the CM template I would use this:

```yaml
{{- range $cmp_plugin, $cmp_plugin_config := .Values.configs.cmp.plugins }}
  {{ $cmp_plugin }}.yaml: |
     kind: ConfigManagementPlugin
     metadata:
       name: {{ $cmp_plugin }}
     spec:
       {{- toYaml $cmp_plugin_config | nindent 6 }}
{{- end }}
```

I think this is less verbose and more readable as YAML instead of multiline string. WDYT @mkilchhofer @jmeridth

Reply:
@pdrastil I agree with this. The YAML this way is much more readable and clean as opposed to the multi-line string I added. Let me know if you'd like me to implement this or if you would like to chip in yourself.

Reply:
@pdrastil I have resolved the things you have highlighted. Let me know if you want me to implement this one or if you would like to chip in yourself.